scripts/qa_crowbarsetup.sh

```diff
@@ -6424,12 +6466,19 @@ function onadmin_batch
     fi
 
     if grep -q "barclamp: octavia" ${scenario}; then
-        # Need neutron deployed first to create the Octavia management network
-        exclude="$exclude --exclude octavia"
+        if [[ $want_octavia_network != 1 ]]; then
+            # Need neutron deployed first to create the Octavia management network
+            exclude="$exclude --exclude octavia"
+        else
+            # If we rely on network.json to configure the Octavia management network,
+            # we can set up the Octavia certs now
+            get_novacontroller
+            setup_octavia_cert /etc/octavia/certs
+        fi
     fi
 
     safely crowbar batch $exclude --timeout 3600 build ${scenario}
-    if grep -q "barclamp: octavia" ${scenario}; then
+    if grep -q "barclamp: octavia" ${scenario} && [[ $want_octavia_network != 1 ]]; then
         get_novacontroller
         setup_octavia_cert /etc/octavia/certs
```

Comment:
is this done in line 6475 already?

Reply:
The opposite, actually. The story behind this silly looking condition is we have two situations:
 - either we add the `octavia` network to network json (`want_octavia_network=1`) and the Octavia barclamp creates the `lb-mgmt-net` network automatically. In this case, we can deploy all the barclamps in one shot.
 - or we don't (`want_octavia_network=0`), in which case we must create the `lb-mgmt-net` network manually (i.e. call the function at line 4194) before we deploy Octavia, but after we deploy Neutron. In this case, we have to deploy everything _except Octavia_, then create the network, then deploy Octavia by itself

This is to ensure we have logic in the automation scripts to cover both scenarios: that in which the customer wants to use the recommended automated way of configuring the `lb-mgmt-net` network by relying on the information supplied in network.json, the other in which customers will provide their own custom `lb-mgmt-net` network